enos/modules/verify_secrets_engines/scripts/kv-pki-verify-certificates.sh

```diff
@@ -0,0 +1,48 @@
+#!/usr/bin/env bash
+# Copyright (c) HashiCorp, Inc.
+# SPDX-License-Identifier: BUSL-1.1
+
+set -e
+
+fail() {
+  echo "$1" 1>&2
+  exit 1
+}
+
+[[ -z "$MOUNT" ]] && fail "MOUNT env variable has not been set"
+[[ -z "$VAULT_ADDR" ]] && fail "VAULT_ADDR env variable has not been set"
+[[ -z "$VAULT_INSTALL_DIR" ]] && fail "VAULT_INSTALL_DIR env variable has not been set"
+[[ -z "$VAULT_TOKEN" ]] && fail "VAULT_TOKEN env variable has not been set"
+[[ -z "$COMMON_NAME" ]] && fail "COMMON_NAME env variable has not been set"
+[[ -z "$TTL" ]] && fail "TTL env variable has not been set"
+[[ -z "$TMP_TEST_RESULTS" ]] && fail "TMP_TEST_RESULTS env variable has not been set"
+
+binpath=${VAULT_INSTALL_DIR}/vault
+test -x "$binpath" || fail "unable to locate vault binary at $binpath" || fail "The certificate appears to be improperly configured or contains errors"
+export VAULT_FORMAT=json
+
+# Getting Certificates
+VAULT_CERTS=$("$binpath" list -format=json "${MOUNT}/certs" | jq -r '.[]')
+
+# Verify List Certificate
+[[ -z "$VAULT_CERTS" ]] && fail "VAULT_CERTS should include vault certificates"
+
+# Verifying Certificates
+for CERT in $VAULT_CERTS; do
```

Comment:
We should probably also verify the contents of each cert that we create. For that we'll need to output the body of each from `create` and then feed the values into `read`. You might want to base64 encode and decode them to make passing the values easier. We do things like that in the various pkcs#11 seal modules if you want to see prior art.

Reply:
updated script to check on the contents like the key, subject, and issuer